### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by mohisa302**
###  Views
In this PR I:

- Implemented the design from the sneak peek wireframes
- Each URL view image in the following:
  - `http://127.0.0.1:3000/users`
  
    image
 
  -` http://127.0.0.1:3000/users/2`
    image

  - `http://127.0.0.1:3000/users/2/posts`
  
   image
   image

  - `http://127.0.0.1:3000/users/2/posts/4`
    image

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,19 @@
...
+  <body>
+<div>
+<% @users.each do |user| %>
+<section class="rounded-lg p-8 flex items-center">
+    <img class="border" src="<%= user.photo %>">
+    <div class="ml-4 text-center flex inform border">
+      <h2 class="font-bold mr-4 mb-2 header-font"><%= user.name %></h2>
+      <p class="text-red-600 par">Number of posts: <%= user.posts_counter %></p>
+    </div>
+    <!-- display user's posts, comments, likes, etc. -->
```
Comment (MeqdamAlqudah) [inline app/views/users/index.html.erb]: - [ ] Please Clicking on each user should take the user to their show page , currently there is no link in the user index page for each user please fix this , so the users can navigate easly through your app.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,25 @@
...
+</section>
+<% @user.recent_posts.each do |post| %>
+<section class="rounded-lg p-8 items-center border"">
+    <h3 class="font-bold">Post #<%= post.id %></h3>
+        <p class="mt-4"><%= post.text %></p>
+    <h5 class="text-red-900 font-bold mt-4 text-right"> comments: <%= post.comments_counter %>, likes: <%= post.likes_counter %></h5>
+    <!-- display user's posts, comments, likes, etc. -->
+</section>
+<% end %>
+<button class="border p-6 font-bold" style="width:20%; margin-left:40%;">See all posts</button>
```
Comment (MeqdamAlqudah) [inline app/views/users/show.html.erb]: - [ ] Please **See all Post** links should opens user's post index , currently this button doesn't do anything,

File: app/views/posts/show.html.erb [hunk 3]
```diff
@@ -1 +1,20 @@
...
+        <h5 class="text-red-900 font-bold mt-4 text-right"> comments: <%= @post.comments_counter %>, likes: <%= @post.likes_counter %></h5>
+      </div>
+      <p class=""><%= @post.text %> </p>
+    </div>
+ <% if @post.recent_comments.any? %>
+      <% @post.recent_comments.each do |comment| %>
+       <div class="border mt-4">
+          <p class="font-bold" style="padding: 2rem"> <%= comment.author.name %>: Comment <%= comment.id %> <p>
+        </div>
+      <% end %>
```
Comment (MeqdamAlqudah) [inline app/views/posts/show.html.erb]: - [ ] Please currently the user can see the comment id and the comment writer name , but he can't see the text of the comment please add it so the user can see the comment info .

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1 +1,25 @@
...
+<section class="rounded-lg border p-8 items-center">
+        <h3 class="font-bold">Bio</h3>
+        <p class="mt-4">
+        <%= @user.bio %>
+        </p>
+</section>
+<% @user.recent_posts.each do |post| %>
+<section class="rounded-lg p-8 items-center border"">
+    <h3 class="font-bold">Post #<%= post.id %></h3>
+        <p class="mt-4"><%= post.text %></p>
```
Comment (MeqdamAlqudah) [inline app/views/users/show.html.erb:19]: - [ ] Please currently the comments box doesn't match the design , please fix it ,the comments should be in a one box I see that each is in different box in your code
 
| your Design | original|
| ------------------ | ---------- |
| image | image |

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @mohisa302  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image


## Highlights
:+1: adding users
:+1: listing posts and comments 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- add links to the post index page the post should send you to the post show page and so on 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1 +1,23 @@
...
+      <div class="flex justify-between">
+        <h3 class="font-bold">Post #1 by <%= @user.name %></h3>
+        <h5 class="text-red-900 font-bold mt-4 text-right"> comments: <%= @post.comments_counter %>, likes: <%= @post.likes_counter %></h5>
+      </div>
+      <p class=""><%= @post.text %> </p>
+    </div>
+ <% if @post.recent_comments.any? %>
+       <div class="border mt-4" style="padding: 2rem">
+      <% @post.recent_comments.each do |comment| %>
+      <div>
```
Comment (m05e5) [inline app/views/posts/show.html.erb]: - [ ] Great job but as part of the requirements, the `show` `post` page is supposed to display all the comments, not just the first 5. Please kindly fix it

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @mohisa302 , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
**Hi @mohisa302 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge:shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights


✔️ Clean Codebase
✔️ No linters error

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Review by m05e5 — APPROVED** [review]
**Hi @mohisa302 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge:shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights


✔️ Clean Codebase
✔️ No linters error

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Review by m05e5 — APPROVED** [review]
**Hi @mohisa302 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge:shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights


✔️ Clean Codebase
✔️ No linters error

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**